### PR TITLE
Remove webpack and unused packages; improve docs; work on github actions 

### PR DESCRIPTION
**Pull request by pvspain**
# Summary

- Remove `webpack` and substituted `vite` for remaining actions.
- Remove all unused `devDependencies` packages (194 packages lighter)
- Update `main.yml` to deploy dev/ demo to GitHub Pages.
- Update `main.yml` to use GitHub official actions.
- Add MacOS bootstrap-demo script to fetch and
    build all tools and crosswords-js package, run `dev/` demo, 
    then cleanup on vanilla MacBook.
- Extract git hook script from `README.md` and add package `qa:install` action  
  -  extend with production assets build as suggested by @mmkal 
- `README` work on QA and workflow advice
- Add combined Linux/MacOS environment bootstrap script

# Details

## modified:   .eslintrc.yml
    Remove webpack ref
## modified:   .github/workflows/main.yml
    Deploy demo site from dev/
    Switch to github official actions for
    uploading artifacts and deploying to Pages
    Add new Node LTS version (20.x)
## modified:   .github/workflows/pull-request.yml
    Remove hardcoded ubuntu version
    Add new Node LTS version (20.x)
## modified:   .gitignore
    Ignore rebase artifacts
## new file:   bin/demo-macos.sh
    Bootstrap MacOS tools and packages, run dev/demo, then cleanup.
    Repoint repo refs to dwmkerr
## modified:   cspell.json
    more words
## modified:   README.md
    Quickstart reference changed to package pulled from hub (npmjs.com)
    Move documentation section to top of file
    De-nest sections under Quickstart
    Re-do "Setting up your dev environment"
        - add section on triangular workflow
        - specific bootstrap instructions for Linux/MacOS/Windows
    Re-do "Quality assurance"
        - Change pre-commit heredoc to action "qa:install"
        - Reference production assets building under "Manual checks"
## renamed:    bin/bootstrap-linux.sh -> bin/bootstrap-posixish.sh
    Combine instructions for Linux/MacOS
## new file:   bin/install-githook.sh
    Add script to install QA hooks - used by action "qa:install"
## modified:   bin/update-package.sh
    Add "npm ci" step
## new file:   dist/crosswords.umd.cjs
    Rebuilt via 'npm build'
## modified:   package-lock.json
    Rebuilt via 'npm ci'
## modified:   package.json
    Switch from yaml -> js-yaml
    Removed webpack
    Removed unused packages (determined by 'npx depcheck')
    Change build action - vite replaces lessc call
    Change action - dev:prod -> dev:preview
    Add action qa:install to install git hook
    Drop action start:angular - broken; moving to separate repo
## modified:   src/crossword-controller.mjs
    Update location of newCellElement() in comments
## modified:   src/crossword-model.mjs
    Change from yaml -> js-yaml
## renamed:    dev/crosswords.js -> src/crosswords.js
    Add ../style/crosswords.less import to get
    vite to transpile src/crosswords.less -> dist/crosswords.css
## modified:   dev/index.js
    Fxup script location
## modified:   test/crossword-model.specs.mjs
    Add bletchley to tested crosswords
    Refactor test on return value from newCrosswordModel()
    - null on failure, and model object otherwise
## new file:   vite.config.js
    Setup config for package 'build' action
    Add 'rollupOptions' hack to rename css file generated
    in src/crosswords.js
## new file:   .githook-pre-commit
    Extract script from README.md
    Add steps to build and stage distribution assets
## modified:   docs/crossword-styling.md
    Add script location for usage of --row-count, --column-count
    Add note for vite generation of CSS assets
## modified:   pull-request-config.sh
    Add vars for merge rollback location - process now includes
    fast-forward commits for simpler commit chains
## modified:   src/crossword-cluesview.mjs, src/crossword-gridview.mjs
    Replace multiple addClass() calls with an addClasses() call
## modified:   src/crossword-model.mjs
    Switch from yaml -> js-yaml
## modified:   style/crosswords.less
    Resolve lint warnings - duplicate rules
    Sort cell sections to match layering order in DOM
    Amalgamate partial sections for same-names
## modified:   style/cwdimensions.less
    Add code location for JS usage of --row-count, --column-count
## modified:   style/crosswords.less, dev/index.html, dev/index.js, dev/index.less
    nfc
## deleted:    webpack.config.js, jsconfig.json, pull-request.config
    Unused
## deleted:    makefile
    Unused now. Using dev/ demo for dwmkerr.github.io
## deleted:    test/index.specs.mjs
    TODO: replace with tests on  src/crosswords.js
## deleted:    src/index.mjs
    src/crosswords.js contains package exports

### CODE REVIEW

File: style/crosswords.less [hunk 1]
```diff
@@ -8,40 +8,25 @@
   grid-template-rows: repeat(var(--row-count), var(--grid-cell-size));
   grid-template-columns: repeat(var(--column-count), var(--grid-cell-size));
 
-  input {
```
Comment (mmkal) [inline style/crosswords.less:11]: Are the changes here just moves or do they change how it looks? Should I test out this branch and make sure it still looks ok on my app?

**Review by mmkal — APPROVED** [review]
LGTM! Small question about styles

File: style/crosswords.less (hunk 1 above)
Comment (pvspain) [reply style/crosswords.less:11]: Hi @mmkal, no behavioural changes intended. Just addressing linting issues and tweaking section order to reflect DOM order.

Please test for behavioural changes!

I'm looking over your PR now. Writing some background for how ids are used in the code!

**Comment by pvspain** [comment]
Hi @dwmkerr, can we merge this PR please? I'm hoping it will fix the demo site.

**Comment by dwmkerr** [comment]
Sure @pvspain any chance you can fix the conflicts and I'll then merge in?

**Comment by pvspain** [comment]
> Sure @pvspain any chance you can fix the conflicts and I'll then merge in?

Fixed now. Thanks @dwmkerr

**Comment by pvspain** [comment]
All done Dave - seasons greetings BTW!
There' a couple of other PRS waiting for review  - resolutions for Issues
#43 and #53.

Cheers, Paul.

**Comment by dwmkerr** [comment]
Amazing, thanks PauL!

**Pull request merged**